syntax = "proto3";

package chat;

service Heartbeat {
  rpc SendHeartbeat(HeartbeatRequest) returns (HeartbeatResponse);
  rpc SwitchServer(SwitchRequest) returns (SwitchResponse);
}

message HeartbeatRequest {
  string server_id = 1;
  uint32 server_port = 2;
}

message HeartbeatResponse {
  bool received = 1;
}

message SwitchRequest {
  string new_primary = 1;
  uint32 server_port = 2;
}

message SwitchResponse {
  bool success = 1;
}